pkgs/build-support/fetchpatch/default.nix

```diff
@@ -27,57 +31,63 @@ let
   };
 in let
   inherit (args') stripLen extraPrefix;
-in
-lib.throwIfNot (excludes == [] || includes == [])
-  "fetchpatch: cannot use excludes and includes simultaneously"
-fetchurl ({
-  postFetch = ''
-    tmpfile="$TMPDIR/patch"
+in let
+  fetch = if !condition then ""
+  else if builtins.isPath path then path
+  else fetchurl ({
```

Comment:
So it's not a fetcher anymore. It's a more general function that should have a general name.
When it's a patch from file, you could still return a package-like object that contains the added metadata.